### PR TITLE
Check dropMatrixMessagesAfterSecs just before sending to IRC

### PR DESCRIPTION
**Pull request by Kegsay**
This is to handle cases where it is set to drop messages after
5 minutes, the message arrives at 4m30s and it takes 1 minute
to go through the bridge (establish connection, join channel, etc).
Previously, the message would go through, 5m30s after the initial
send. Now, it won't go through.

Fixes issues mentioned in #388 by @justjanne

### CODE REVIEW

File: lib/irc/IrcServer.js [hunk 1]
```diff
@@ -11,8 +11,12 @@ var log = logging.get("IrcServer");
  * @constructor
  * @param {string} domain : The IRC network address
  * @param {Object} serverConfig : The config options for this network.
+ * @param {string} homeserverDomain : The domain of the homeserver
+ * e.g "matrix.org"
+ * @param {Number} expiryTimeSeconds : The max number of seconds a
+ * message can be in order for it to be sent to this server.
```
Comment (lukebarnard1) [inline lib/irc/IrcServer.js]: I'm not sure this makes sense. What does "a message can be" mean?

File: lib/irc/IrcServer.js [hunk 2]
```diff
@@ -22,6 +26,16 @@ function IrcServer(domain, serverConfig, homeserverDomain) {
     }
     this._addresses.push(domain);
     this._homeserverDomain = homeserverDomain;
+    this._expiryTimeSeconds = expiryTimeSeconds;
+}
+
+/**
+ * Get the max number of seconds a message can be before it is dropped from
```
Comment (lukebarnard1) [inline lib/irc/IrcServer.js]: again, I'm not sure this is phrased well, see above.

File: spec/integ/matrix-to-irc.spec.js [hunk 3]
```diff
@@ -548,6 +550,47 @@ describe("Matrix-to-IRC message bridging with media URL and drop time", function
...
+        let said = false;
+        env.ircMock._whenClient(roomMapping.server, testUser2.nick, "say",
+        function(client, channel, text) {
+            said = true;
+        });
+
+        let connected = false;
+        env.ircMock._whenClient(roomMapping.server, testUser2.nick, "connect",
+        function(client, cb) {
+            jasmine.clock().tick(20 * 1000); // advance 20s to take it over the drop time
```
Comment (lukebarnard1) [inline spec/integ/matrix-to-irc.spec.js]: I would make it obvious that "drop time" is actually `dropMatrixMessagesAfterSecs`

Comment (Kegsay) [inline spec/integ/matrix-to-irc.spec.js]: Done. https://github.com/matrix-org/matrix-appservice-irc/pull/412/commits/e02ee95691f1f4407bc6e2a8735aa148c834f14b

File: lib/irc/IrcServer.js (hunk 2 above)
Comment (Kegsay) [inline lib/irc/IrcServer.js]: Done. https://github.com/matrix-org/matrix-appservice-irc/pull/412/commits/e02ee95691f1f4407bc6e2a8735aa148c834f14b

File: lib/irc/IrcServer.js (hunk 1 above)
Comment (Kegsay) [inline lib/irc/IrcServer.js]: Done. https://github.com/matrix-org/matrix-appservice-irc/pull/412/commits/e02ee95691f1f4407bc6e2a8735aa148c834f14b

**Comment by justjanne** [comment]
Aren't you still missing the word "old"? As in, `the maximum number of seconds a message can be old before it is dropped`, or `the maximum age of a message before it is dropped`

**Comment by Kegsay** [comment]
Yep, I've re-phrased with "old" already:

```
 + * @param {Number} expiryTimeSeconds : How old a matrix message can be
 + * before it is considered 'expired' and not sent to IRC. If 0, messages
 + * will never expire.
```

**Pull request merged**